scripts/ci/test_reborn_changed_coverage.py

```
@@ -63,6 +63,218 @@ def test_reborn_workflow_runs_strict_gate_and_preserves_machine_report(self):
+        self.assertIn(
+            '"preexisting_uncovered": preexisting_uncovered',
+            source,
+            "the machine report must serialise the complete, uncapped list",
+        )
+
+    def test_an_lcov_with_no_da_records_is_refused_as_base_coverage(self):
+        with tempfile.TemporaryDirectory() as temp:
+            empty = Path(temp) / "base.lcov"
+            empty.write_text("TN:\nSF:/nowhere.rs\nend_of_record\n", encoding="utf-8")
+
+            with self.assertRaises(gate.BaseCoverageUnavailable):
```

Ruff (0.16.0)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)